### PR TITLE
feat: Adds hybrid search options contract for multisearch

### LINKED ISSUE #676: Hybrid search options missing for new federated search feature in v1.10 Meilisearch
**Description**
Hybrid search options missing for new federated search feature in v1.10 Meilisearch

**Expected behavior**
Hybrid search options should be available.
['semanticRatio'=> 0.8,'embedder' => 'manual',];

**Current behavior**
Missing in PHP src/Contracts/SearchQuery.php

**Environment (please complete the following information):**
- Meilisearch version: v1.10 / v1.10.1
- meilisearch-php version: v1.10

### PR DESCRIPTION
**Pull request by apozeus**
# Pull Request

## Related issue
Fixes #676

## What does this PR do?
Adds hybrid search options contract for multisearch

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by norkunas** [comment]
@apozeus why you opened same code changes with two separate pull requests? 🙂

**Comment by apozeus** [comment]
> @apozeus why you opened same code changes with two separate pull requests? 🙂

Oh my mistake. i thought i needed to open an issue then make a PR.

**Comment by norkunas** [comment]
> > @apozeus why you opened same code changes with two separate pull requests? 🙂
> 
> Oh my mistake. i thought i needed to open an issue then make a PR.

Well you could open an issue and then link in the previous PR description :slightly_smiling_face: 
As this is I think a priority bug/feature and easy to fix it, at least I'd merge also without opening the issue.

By the way, I've left some comments in your previous PR.

**Comment by apozeus** [comment]
> > > @apozeus why you opened same code changes with two separate pull requests? 🙂
> > 
> > 
> > Oh my mistake. i thought i needed to open an issue then make a PR.
> 
> Well you could open an issue and then link in the previous PR description 🙂 As this is I think a priority bug/feature and easy to fix it, at least I'd merge also without opening the issue.
> 
> By the way, I've left some comments in your previous PR.

Commited your comments.

**Comment by norkunas** [comment]
Ok so as you left this one open, then please apply all code change suggestions from previous PR

**Comment by apozeus** [comment]
> Ok so as you left this one open, then please apply all code change suggestions from previous PR

it's all applied.

**Comment by norkunas** [comment]
> > Ok so as you left this one open, then please apply all code change suggestions from previous PR
> 
> it's all applied.

Nope :) I don't see it, maybe you forgot to commit?

File: src/Contracts/HybridSearchOptions.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Meilisearch\Contracts;
+
+class HybridSearchOptions
+{
+    private ?float $semanticRatio = null;
+    private ?string $embedder = null;
```
Comment (norkunas) [inline src/Contracts/HybridSearchOptions.php:10]:
```suggestion

    /**
     * @var non-empty-string|null
     */
    private ?string $embedder = null;
```

File: src/Contracts/HybridSearchOptions.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public function setSemanticRatio(float $ratio): HybridSearchOptions
+    {
+        $this->semanticRatio = $ratio;
+
+        return $this;
+    }
+
+    /**
+     * @param non-empty-string
```
Comment (norkunas) [inline src/Contracts/HybridSearchOptions.php:20]:
```suggestion
     * @param non-empty-string $embedder
```

File: src/Contracts/HybridSearchOptions.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    public function setEmbedder(string $embedder): HybridSearchOptions
+    {
+        $this->embedder = $embedder;
+
+        return $this;
+    }
+
+    public function toArray(): array
```
Comment (norkunas) [inline src/Contracts/HybridSearchOptions.php:29]:
```suggestion
    /**
     * @return array{
     *     semanticRatio?: float,
     *     embedder?: non-empty-string
     * }
     */
    public function toArray(): array
```

**Review by norkunas — APPROVED** [review]

**Comment by brunoocasali** [comment]
bors merge

**Pull request merged**